### ISSUE TITLE
Try Other Architectures

### ISSUE THREAD

**Issue by wingman-jr-addon** [wingman-jr-addon/wingman_jr#64]
Now that I have about enough data to train from scratch and a baseline of about 72-73% raw accuracy, I can start to explore other architectures. While nothing terribly earth-shattering, there have been a number of improvements (mostly regarding efficiency).

EfficientNet (#58) is being tracked in its own thread, but does not seem to be giving any particular efficiency/accuracy improvement.

One other architecture that is straightforward to try is [IoTNet](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6960729/). This provides a family of networks to try. I have made some initial progress here and can comment when the results are a bit more complete.

**Comment by wingman-jr-addon** [comment]
Update on EfficientNet - I am seeing some progress there actually now using fine tunes.

I played around with IoTNet quite a bit, tried many different variants but at the end of the day I was not able to improve overall accuracy.

As a general observation about speed - while it's true that raw prediction speed makes a difference, a vast chunk of the work is actually in decoding and resizing the image, so simply having a bit smaller network does not translate into automatic gains. This has an impact on network choices. However, startup time will likely correlate closely to the number of architecture parameters.

**Comment by issuemaker2** [comment]
[dicenet](https://arxiv.org/pdf/1906.03516.pdf)([github repo](https://github.com/sacmehta/EdgeNets)) seems to worth looking at it, though currently there's no tensorflow implementation available.

**Comment by wingman-jr-addon** [comment]
That dicenet one looks interesting. I'm going to have keep an eye on that.

**Comment by wingman-jr-addon** [comment]
(Also, if you have any brilliant ideas on how that might actually look in getting ported to what Tensorflow.js supports, I'd love to hear them. :) I was able to pull over some simple stuff to get EfficientNet working but a whole new conv might get interesting?)

**Comment by wingman-jr-addon** [comment]
There was a nice summary [over on machinethink.net](https://machinethink.net/blog/mobile-architectures/) of some more recent lightweight architectures. (@issuemaker2 you may be interested too - a few on that list were new to me). I was giving EfficientLite a bit of a shot as I found a [Keras port](https://github.com/qubvel/efficientnet/issues/104); so far no dice with finetuning using my MobileNetV2 hyperparams. Maybe my normalization is off.....

**Comment by wingman-jr-addon** [comment]
Yep, it was normalization. Finetune got similar accuracy, BUT the fused ROC output was better. Dropping it onto master with #91 along with better ROC work.

**Comment by wingman-jr-addon** [comment]
The original intent of this issue has been fixed. I will still keep my eyes peeled for new things like advancements in Vision Transformers (#130) and semi- and unsupervised learning approaches (#124) but will close this issue.
